widgets/modules/EnsEMBL/Web/Apache/Static.pm

```diff
@@ -0,0 +1,36 @@
+package EnsEMBL::Web::Apache::Static;
+
+use strict;
+use warnings;
+
+sub add_caching_headers {
+  my $r = shift;
+  my $uri = $r->uri;
+
+  if (should_skip_caching($uri)) {
+    $r->headers_out->set('Cache-Control'  => 'no-store, max-age=0');
+  } else {
+    # default behaviour
+    my $thirty_days = 60 * 60 * 24 * 30;
+    $r->headers_out->set('Cache-Control'  => 'max-age=' . $thirty_days);
+    $r->headers_out->set('Expires'        => HTTP::Date::time2str(time + $thirty_days));
+  }
+}
+
+sub should_skip_caching {
+  my $uri = shift;
+  my @patterns = (
```

Comment:
why an array? If it is just one string regex, no need for an array and this will get rid of the foreach below.

Reply:
i will write this like this:
{code}
my $uri = shift;

if($uri =~ qr/alphafold\/.*\.js$/) {
 return 1;
} else {
 return 0;
}
 {code}

Reply:
> why an array?

Arrays are easy to extend. If we refactor the pdbe widget in a similar manner (there are plans to do so); we would also add the pdbe folder to the patterns list. Are there any problems with using the array?

Reply:
array makes the code slower. if takes less memory not that it makes a big difference here but also the code is cleaner without the array (easier to understand). if we decide to add pdbe we can just add it to the regex.

Reply:
i dont think you need a function to should_skip_caching. just have the code like this:
`if($uri =~ /alphafold\/.*\.js$/) {` instead of the function. By having this in the plugin, you are already expecting to put whatever is needed to skip the caching in the function. Creating another function here is just adding complexity to the code.

Reply:
In code review, we decided to use this method bcs it is more readable... with some minor additions like below to create the match strings.
(
'aplhafold',
pdbe'
) => () => { to return a string for matching }

Reply:
With some modifications, implemented here: https://github.com/Ensembl/public-plugins/blob/7f9fbb0a27df59b6c6aae725d11449c12f13d89c/widgets/modules/EnsEMBL/Web/Apache/Static.pm